### ISSUE TITLE
Crashes when using Gradle build system (Unity 2018.3.0)

### ISSUE THREAD

**Issue by rebuilder17** [playgameservices/play-games-plugin-for-unity#2402]
When I build my apk with Gradle build system (Build Settings > Build System as "Gradle") the whole build process goes just ok but get some errors on runtime.

```
E	6736		GamesNativeSDK	Exception in dalvik/system/DexClassLoader.loadClass: java.lang.IncompatibleClassChangeError: Class com.google.android.gms.games.NativeSdkCallbacks implements non-interface class com.google.android.gms.nearby.connection.Connections$ConnectionRequestListener (declaration of 'com.google.android.gms.games.NativeSdkCallbacks' appears in /data/data/( ... my bundle name ... )/app_.gpg.classloader/d75c8e1bbeab9e1ddffb9d332585db71.jar).
F	6736		zygote	java_vm_ext.cc:504] JNI DETECTED ERROR IN APPLICATION: java_class == null
F	6736		zygote	java_vm_ext.cc:504]     in call to RegisterNatives
F	6736		zygote	java_vm_ext.cc:504]     from java.lang.Object com.unity3d.player.ReflectionHelper.nativeProxyInvoke(int, java.lang.String, java.lang.Object[])
```

I don't get this error when using Internal (deprecated) build system.

**Comment by olehkuznetsov** [comment]
Which plugin version do you use?

**Comment by rebuilder17** [comment]
> 
> 
> Which plugin version do you use?

I'm using 0.9.57 now.

**Comment by Pablo-Benitez** [comment]
Happening to me too, building with Unity 2017.3.1f1 using Gradle and plugin version 0.9.58

Dependencies are up to date too (I think):

<img width="376" alt="screen shot 2019-01-20 at 21 39 13" src="https://user-images.githubusercontent.com/46873714/51444762-2c2d1880-1cfc-11e9-9ed8-f000eeb51b0c.png">

**Comment by olehkuznetsov** [comment]
It would help a lot if you try it on a new project without Firebase first. May be it is some incompatibility with a new firebase plugin.

**Comment by rubndsouza** [comment]
Thanks, but unfortunately I don't have Firebase on my project.

**Comment by olehkuznetsov** [comment]
Can you show your current dependencies list as in reply above, please?

**Comment by olehkuznetsov** [comment]
Thank you for the update. Will check how to fix GPGSUpgrader.cs

**Comment by jjurica** [comment]
After I read @rubndsouza 's comment I looked myself at GPGSUpgrader.cs. I do not have the bug thats in the title. However, I was wondering since I have GPGS 9.58, should all the files listed in GPGSUpgrader.cs as obsolete be removed? Because all of the files in "Assets/GooglePlayGames/Platforms/Android/Gms/Games" are still in my project. I don't remember the first version I installed, but the last one I upgraded from was 9.50. Does having these old files cause any issues or can I ignore them?
